TheaterSchedule.BLL/Services/ImageService.cs

```diff
@@ -1,16 +1,188 @@
+                return "text/plain";
+            }
+        }
+
+        public async Task<ImageBase64DTO> LoadPerformanceMainImageBase64Async(int id)
+        {
+            ImageBase64DTO imageDTO = null;
+            byte[] image = await TryAddPerformanceImageToCacheAsync(id);
+
+            if (image != null)
+            {
+                await Task.Run(() =>
+                {
+                    imageDTO = new ImageBase64DTO()
+                    {
+                        Image = ImageToBase64(image),
+                        ImageFormat = GeMimeTypeFromImageByteArray(image)
+                    };
+                });
+            }
+
+            return imageDTO;
+        }
+
+        public async Task<ImageBytesDTO> LoadPerformanceMainImageBytesAsync(int id)
+        {
+            ImageBytesDTO imageDTO = null;
+            byte[] image = await TryAddPerformanceImageToCacheAsync(id);
+
+            if (image != null)
+            {
+                await Task.Run(() =>
+                {
+                    imageDTO = new ImageBytesDTO()
+                    {
+                        Image = image,
+                        ImageFormat = GeMimeTypeFromImageByteArray(image)
+                    };
+                });
+            }
+
+            return imageDTO;
+        }
+
+        public async Task<List<ImageBase64DTO>> LoadPerformanceGalleryBase64Async(int id)
+        {
+            List<ImageBase64DTO> imagesDTO = new List<ImageBase64DTO>();
+            IEnumerable<byte[]> images = await TryAddGalleryImagesToCacheAsync(id);
+
+            await Task.Run(() =>
+            {
+                foreach (var image in images)
+                {
+                    imagesDTO.Add(new ImageBase64DTO()
+                    {
+                        Image = ImageToBase64(image),
+                        ImageFormat = GeMimeTypeFromImageByteArray(image)
+                    });
+                }
+            });
+
+            return imagesDTO;
+        }
+
+        public async Task<List<ImageBytesDTO>> LoadPerformanceGalleryBytesAsync(int id)
+        {
+            List<ImageBytesDTO> imagesDTO = new List<ImageBytesDTO>();
+            IEnumerable<byte[]> images = await TryAddGalleryImagesToCacheAsync(id);
+
+            await Task.Run(() =>
+            {
+                foreach (var image in images)
+                {
+                    imagesDTO.Add(new ImageBytesDTO()
+                    {
+                        Image = image,
+                        ImageFormat = GeMimeTypeFromImageByteArray(image)
+                    });
+                }
+            });
+
+            return imagesDTO;
+        }
+
+
+        public async Task<IEnumerable<byte[]>> TryAddGalleryImagesToCacheAsync(int performanceId)
+        {
+            bool needUpdateCache = false;
+            IEnumerable<byte[]> images = new List<byte[]>();
+            int imgsNumber = await imageRepository.GetGalleryImagesCountAsync(performanceId);

```

Comment:
no need get count
just get List of images and iterate through it

Reply:
Yes, but in this case caching will be useless

Reply:
I think it isa bit faster to get rows count that data